File: ScoutTrack/ScoutTrack.UI/scouttrack_desktop/lib/models/equipment.dart
import 'package:json_annotation/json_annotation.dart';

part 'equipment.g.dart';

@JsonSerializable()
class Equipment {
  final int id;
  final String name;
  final String description;
  final bool isGlobal;
  final int? createdByTroopId;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Equipment({
    this.id = 0,
    this.name = '',
    this.description = '',
    this.isGlobal = true,
    this.createdByTroopId,
    required this.createdAt,
    this.updatedAt,
  });

  factory Equipment.fromJson(Map<String, dynamic> json) =>
      _$EquipmentFromJson(json);

  Map<String, dynamic> toJson() => _$EquipmentToJson(this);
}

File: ScoutTrack/ScoutTrack.Services/Database/Entities/Equipment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ScoutTrack.Services.Database.Entities
{
    public class Equipment
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;

        public bool IsGlobal { get; set; } = true;
        public int? CreatedByTroopId { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime? UpdatedAt { get; set; }
        public ICollection<ActivityEquipment> ActivityEquipments { get; set; } = new List<ActivityEquipment>();
    }
}
